bitcoin/src/blockdata/witness.rs

```diff
@@ -95,16 +100,37 @@ impl Decodable for Witness {
                 cursor += element_size;
```

Comment:
Oh, I was wondering why the hell are we resizing the `Vec` and thus writing a bunch of zeroes every time and making the code crazy complicated. The reason is we accept `Read` instead of `BufRead` which I previously suggested. So we're wasting performance. I will think about improving this.